libecs/source/Core/Data/type.cxx

```diff
@@ -64,4 +64,18 @@ data::Value Type::defaultValue() const
         },
         static_cast<const Type::variant &>(*this));
 }
+
+bool Type::isSubsetOf(const Type &other) const
+{
+    if (const RecordType *record = std::get_if<RecordType>(this)) {
```

Comment:
if *this == other, does that mean this test will fail ?
If it doesn't, perhaps move the test at line 79 up in order to exit early.

Reply:
The function can return true even when *this == other is false !